docs: Refactor CRD docs and flags description (#2506)
kyma-project/lifecycle-manager: docs/contributor/08-kcp-skr-synchronization.md

The v1beta2 API introduces three groups of modules:

* **Default** modules that are synchronized by default.

> * **Internal** modules that are synchronized per-cluster only if the corresponding Kyma CR contains the `operator.kyma-project.io/internal` label. A ModuleTemplate CR is `internal` if the `operator.kyma-project.io/internal` label on it is set to `true` by module author.
> * **Beta** modules that are synchronized per-cluster only if the corresponding Kyma CR contains the `operator.kyma-project.io/beta` label. A ModuleTemplate CR is `beta` if the `operator.kyma-project.io/beta` label on it is set to `true` by module author.

Reviewer: ```suggestion
* **Internal** modules that are synchronized per cluster only if the corresponding Kyma CR has the `operator.kyma-project.io/internal` label. A ModuleTemplate CR is `internal` if the module team sets the `operator.kyma-project.io/internal` label to `true`.
* **Beta** modules that are synchronized per cluster only if the corresponding Kyma CR contains the `operator.kyma-project.io/beta` label. A ModuleTemplate CR is `beta` if the module team sets the `operator.kyma-project.io/beta` label to `true`.
```
